src/emulator.py

```diff
@@ -87,58 +88,49 @@ def get_status() -> dict:
 
 
 def start(version: str, wipe: bool) -> None:
-    global proc
     global version_running
+    global EMULATOR
 
-    if proc is not None:
+    if EMULATOR is not None:
         log(
             f"Before starting a new emulator - version {version}, "
             f"killing the already running one - version {version_running}",
             "red",
         )
         stop()
 
-    path = ROOT_DIR / "src/binaries/firmware/bin"
-
     if version[0] == "2":
-        model_t_profile = "/var/tmp/trezor.flash"
-        if wipe and os.path.exists(model_t_profile):
-            os.remove(model_t_profile)
-
-        command = f"{path}/trezor-emu-core-v{version} -O0 -X heapsize=20M -m main"
+        EMULATOR = CoreEmulator(
+            FIRMWARE_DIR / f"trezor-emu-core-v{version}", profile_dir=FIRMWARE_DIR
+        )
+    elif version[0] == "1":
+        os.environ["TREZOR_OLED_SCALE"] = str(TREZOR_ONE_OLED_SCALE)
+        EMULATOR = LegacyEmulator(
+            FIRMWARE_DIR / f"trezor-emu-legacy-v{version}",
+            profile_dir=FIRMWARE_DIR,
+            extra_args=["-O0"],
```

Comment:
AFAIK the `legacy` emulator does not need nor use this argument. Not sure why it was there originally, perhaps so that you could run `core` and `legacy` with the same command?

Reply:
Thanks for pointing this, I have deleted it. It really could be that previously there was same command for `core` and `legacy`